src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -733,11 +733,20 @@ or ImplicitStackAllocArrayCreationExpressionSyntax
                 .WithPrependedLeadingTrivia(ElasticMarker);
     }
 
+    private static bool HasStructuredTriviaBetweenCloseTokenAndInitializer(InitializerExpressionSyntax initializer)
+    {
+        static bool IsComment(SyntaxTrivia trivia) => trivia.IsKind(SyntaxKind.SingleLineCommentTrivia) || trivia.IsKind(SyntaxKind.MultiLineCommentTrivia);
+        return initializer.OpenBraceToken.GetPreviousToken().TrailingTrivia.Any(IsComment);
+    }
+
     private static bool ShouldReplaceExistingExpressionEntirely(
         SourceText sourceText,
         InitializerExpressionSyntax initializer,
         bool newCollectionIsSingleLine)
     {
+        if (HasStructuredTriviaBetweenCloseTokenAndInitializer(initializer))
```

Comment:
the method name is also super confusing as it references the CloseToken, but is actually checking the OpenBrace of hte initializer.